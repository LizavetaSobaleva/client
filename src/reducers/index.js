import { applyMiddleware, combineReducers } from "redux";
import { configureStore} from "@reduxjs/toolkit"
import { thunk } from "redux-thunk";
import userReducer from "./userReducer";
import fileReducer from "./fileReducer";
import uploadReducer from "./uploadReducer";
import appReducer from "./appReducer";
import { composeWithDevTools } from '@redux-devtools/extension';


const rootReducer = combineReducers({
    user: userReducer,
    files: fileReducer,
    upload: uploadReducer,
    app: appReducer
})

export const store = configureStore({reducer: rootReducer, devTools: composeWithDevTools(applyMiddleware(thunk))})