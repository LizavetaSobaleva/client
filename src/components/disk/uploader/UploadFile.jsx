import React from 'react'
import './uploader.less'
import fileLogo from '../../../assets/img/file.svg'
import ProgressBar from '../../UI/progressBar/ProgressBar'
import close from '../../../assets/img/close.svg'
import SecondaryButton from '../../UI/secondaryButton/SecondaryButton'
import { useDispatch } from 'react-redux'
import { removeUploadFile } from '../../../reducers/uploadReducer'

const UploadFile = ({file}) => {
    const dispatch = useDispatch()


    return (
        <div className='uploadFile'>
            <img src={fileLogo} alt="" className='uploadFile__img'/>
            <div className="uploadFile__content">
                <div className="uploadFile__name">{file.name}</div>
                <ProgressBar progress={file.progress}/>
            </div>
            <SecondaryButton onClick={() => dispatch(removeUploadFile(file.id))}>
                <img src={close} alt='' className='popup__close'/>
            </SecondaryButton>
            

        </div>
    )
}

export default UploadFile