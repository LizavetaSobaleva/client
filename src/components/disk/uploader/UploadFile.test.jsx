import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import UploadFile from './UploadFile';
import { removeUploadFile } from '../../../reducers/uploadReducer';

jest.mock('../../UI/progressBar/ProgressBar', () => ({ progress }) => (
  <div data-testid="progressBar">{progress}%</div>
));

const mockStore = configureStore([]);

const file = {
  id: '1',
  name: 'file1.txt',
  progress: 50
};

let store;

beforeEach(() => {
  store = mockStore({});
  store.dispatch = jest.fn();
});

const renderComponent = () => render(
  <Provider store={store}>
    <UploadFile file={file} />
  </Provider>
);

test('renders UploadFile without crashing', () => {
  const { asFragment } = renderComponent();
  expect(screen.getByTestId('fileName')).toHaveTextContent('file1.txt');
  expect(screen.getByTestId('progressBar')).toHaveTextContent('50%');
  expect(asFragment()).toMatchSnapshot();
});

test('close button triggers removeUploadFile action', () => {
  const { asFragment } = renderComponent();
  fireEvent.click(screen.getByTestId('closeBtn'));
  expect(store.dispatch).toHaveBeenCalledWith(removeUploadFile('1'));
  expect(asFragment()).toMatchSnapshot();
});
