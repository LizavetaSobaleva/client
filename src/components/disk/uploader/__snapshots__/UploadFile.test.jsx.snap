// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`close button triggers removeUploadFile action 1`] = `
<DocumentFragment>
  <div
    class="uploadFile"
  >
    <img
      alt=""
      class="uploadFile__img"
      src="test-file-stub"
    />
    <div
      class="uploadFile__content"
    >
      <div
        class="uploadFile__name"
        data-testid="fileName"
      >
        file1.txt
      </div>
      <div
        data-testid="progressBar"
      >
        50%
      </div>
    </div>
    <button
      class="secondaryButton"
      data-testid="closeBtn"
    >
      <img
        alt=""
        class="popup__close"
        src="test-file-stub"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`renders UploadFile without crashing 1`] = `
<DocumentFragment>
  <div
    class="uploadFile"
  >
    <img
      alt=""
      class="uploadFile__img"
      src="test-file-stub"
    />
    <div
      class="uploadFile__content"
    >
      <div
        class="uploadFile__name"
        data-testid="fileName"
      >
        file1.txt
      </div>
      <div
        data-testid="progressBar"
      >
        50%
      </div>
    </div>
    <button
      class="secondaryButton"
      data-testid="closeBtn"
    >
      <img
        alt=""
        class="popup__close"
        src="test-file-stub"
      />
    </button>
  </div>
</DocumentFragment>
`;
